lib/Widget/Text.php

```diff
@@ -273,8 +276,15 @@ public function getResource($displayId = 0)
 
         // Finalise some JavaScript to run.
         $javaScriptContent = '$(document).ready(function() { ';
-        $javaScriptContent .= '       $("#content").xiboTextRender(options, items); $("body").xiboLayoutScaler(options); $("#content").find("img").xiboImageRender(options); ';
+        
+        // Run on document ready
+        $javaScriptContent .= '     $("body").xiboLayoutScaler(options); $("#content").find("img").xiboImageRender(options); ';
 
+        // Run based only if the element is visible or not
+        $javaScriptContent .= '     const runOnVisible = function() { $("#content").xiboTextRender(options, items); }; ';
```

Comment:
So I think we can apply this logic to other types of Widget, except Embedded/Web Page/Package HTML?